python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -469,24 +473,30 @@ def accessURL(self, thisSE=''):
         """
         _accessURLs = []
         if len(self.subfiles) == 0:
-          self.getReplicas()
-          # If the SE isn't specified return a random choice.
-          if thisSE == '':
-            this_SE = random.choice(self.locations)
-          # If the SE is specified and we got a URL for a replica there, use it.
-          elif thisSE in self.locations:
-            this_SE = thisSE
-          # If the specified SE doesn't have a replica then return another one at random.
-          else:
-             logger.warning('No replica at specified SE for the LFN %s, here is a URL for another replica' % self.lfn)
-             this_SE = random.choice(self.locations) 
-          myurl = execute('getAccessURL("%s" , "%s")' % (self.lfn, this_SE))
-          this_accessURL = myurl['Value']['Successful'][self.lfn]
-          _accessURLs.append(this_accessURL)
+            self.getReplicas()
+            # If the SE isn't specified return a random choice.
+            if thisSE == '':
+                this_SE = random.choice(self.locations)
+            # If the SE is specified and we got a URL for a replica there, use it.
+            elif thisSE in self.locations:
+                this_SE = thisSE
+            # If the specified SE doesn't have a replica then return another one at random.
+            else:
+                logger.warning('No replica at specified SE for the LFN %s, here is a URL for another replica' % self.lfn)
+                this_SE = random.choice(self.locations)
+
+            try:
+                myurl = execute('getAccessURL("%s" , "%s")' % (self.lfn, this_SE))
+            except GangaDiracException as err:
+                raise GangaException("Error requesting URL: %s" % err)
```

Comment:
Here (and elsewhere in this PR) we should avoid raising plain `GangaException`s. That class is intended simply as the root of the inheritance hierarchy, not as something that is actively used. Try finding an appropriate exception type (e.g. `GangaIOError`) and if there isn't one create it.